src/toxcore/dht/daemon_state.rs

```diff
@@ -15,25 +13,96 @@ use toxcore::state_format::old::*;
 use toxcore::binary_io::*;
 use toxcore::dht::kbucket::*;
 
-/// Error that can happen when calling `deserialize_old` of DhtState.
-#[derive(Debug, Fail, PartialEq)]
-pub enum DeserializeError {
-    /// Error indicates that DhtState object can't be parsed.
-    #[fail(display = "Deserialize DhtState error: {:?}, packet: {:?}", error, data)]
+use std::fmt;
+
+use failure::{Backtrace, Context, Fail};
+use nom::{Needed, ErrorKind as NomErrorKind};
+
+/// An error that can occur while serializing/deserializing object
+#[derive(Debug)]
+pub struct SerDeError{
+    ctx: Context<SerDeErrorKind>,
+}
+
+impl SerDeError{
```

Comment:
That looks like DeserializeError not like SerDeError because there is no serialization errors, only deserialization

Reply:
done